src/main/java/org/springframework/integration/kafka/inbound/KafkaInboundChannelAdapter.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.springframework.integration.kafka.inbound;
+
+import kafka.serializer.Decoder;
+import kafka.serializer.DefaultDecoder;
+
+import org.springframework.integration.context.OrderlyShutdownCapable;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.kafka.core.KafkaMessageMetadata;
+import org.springframework.integration.kafka.listener.AbstractDecodingMessageListener;
+import org.springframework.integration.kafka.listener.KafkaMessageListenerContainer;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.util.Assert;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class KafkaInboundChannelAdapter extends MessageProducerSupport implements OrderlyShutdownCapable {
+
+	public static final String KAFKA_MESSAGE_KEY = "kafka.message.key";
+
+	public static final String KAFKA_MESSAGE_TOPIC = "kafka.message.topic";
+
+	public static final String KAFKA_MESSAGE_PARTITION = "kafka.message.partition.id";
+
+	public static final String KAFKA_MESSAGE_OFFSET = "kafka.message.offset";
```

Comment:
Spring Integration headers (now) have the convention `<module>_camelCase`; e.g. kafka_messageOffset - also they should be moved to `KafkaHeaders` (message key and topic already exist over there).